app/mailers/order_mailer.rb

```diff
@@ -4,8 +4,19 @@ class OrderMailer < ApplicationMailer
 
   def received(order)
     @order = order
+    order.line_items.each do |line_item|
+      product = line_item.product
+      product_images = product.images.to_a
 
-    mail to: order.email, subject: 'Pragmatic Store Order Confirmation'
+      attachments.inline["#{product.title}_1"] = product_images.shift.download
+      product_images.each do |image|
+        attachments["#{product.title}_#{image.filename}"] = image.download
```

Comment:
Shouldn't these be inline attachments?
do we really need to download the image, can't we give the path of the image?

Reply:
Extract this in separate method as well

Reply:
No need todownload